base/strings/search.jl

```diff
@@ -194,7 +217,7 @@ function _searchindex(s::ByteArray, t::ByteArray, i::Integer)
     bloom_mask = UInt64(0)
     skip = n - 1
     tlast = _nthbyte(t,n)
-    for j in 1:n
+    for j in firstindex(s):n
```

Comment:
This doesn't look right: `n` may not be the last index.  Make sure you test this function where `s` and/or `t` are offset arrays.